internal/manifests/targetallocator/service.go

```diff
@@ -23,7 +23,7 @@ func Service(params manifests.Params) *corev1.Service {
 		labels["app.kubernetes.io/version"] = "latest"
 	}
 
-	selector := Labels(params.OtelCol, naming.TAService())
+	selector := Labels(params.OtelCol, naming.TAPodDestination(params.OtelCol.Name))
```

Comment:
Thanks for catching and fixing this issue.

Reply:
This could be a constant instead of a naming function. The agent service follow this format - https://github.com/aws/amazon-cloudwatch-agent-operator/blob/main/internal/manifests/collector/service.go#L108

Reply:
The constant is defined here - https://github.com/aws/amazon-cloudwatch-agent-operator/blob/main/internal/manifests/collector/collector.go#L15
We should do something similar for target-allocator as well